aries/aries_vcx_core/src/wallet2/indy_wallet/indy_did_wallet.rs

```diff
@@ -0,0 +1,145 @@
+use async_trait::async_trait;
+use vdrtools::{DidMethod, DidValue, KeyInfo, Locator, MyDidInfo};
+
+use crate::{
+    errors::error::{AriesVcxCoreError, VcxCoreResult},
+    wallet2::{DidData, DidWallet},
+};
+
+#[async_trait]
+impl DidWallet for crate::wallet::indy::IndySdkWallet {
+    async fn create_and_store_my_did(
+        &self,
+        seed: &str,
+        method_name: Option<&str>,
+    ) -> VcxCoreResult<DidData> {
+        let res = Locator::instance()
+            .did_controller
+            .create_and_store_my_did(
+                self.wallet_handle,
+                MyDidInfo {
+                    method_name: method_name.map(|m| DidMethod(m.into())),
+                    seed: Some(seed.into()),
+                    ..MyDidInfo::default()
+                },
+            )
+            .await
+            .map_err::<AriesVcxCoreError, _>(From::from)?;
+
+        Ok(DidData {
+            did: res.0,
+            verkey: res.1,
+        })
+    }
+
+    async fn did_key(&self, did: &str) -> VcxCoreResult<String> {
+        Locator::instance()
+            .did_controller
+            .key_for_local_did(self.wallet_handle, DidValue(did.into()))
+            .await
+            .map_err(From::from)
+    }
+
+    async fn replace_did_key(&self, did: &str, seed: &str) -> VcxCoreResult<String> {
+        let key_info = KeyInfo {
+            seed: Some(seed.into()),
+            ..Default::default()
+        };
+
+        let key = Locator::instance()
+            .did_controller
+            .replace_keys_start(self.wallet_handle, key_info, DidValue(did.into()))
+            .await?;
+
+        Locator::instance()
+            .did_controller
+            .replace_keys_apply(self.wallet_handle, DidValue(did.into()))
+            .await?;
+
+        Ok(key)
+    }
```

Comment:
The higher levels of the stack require these separate. For example `rotate_verkey` in `aries/aries_vcx/src/common/keys.rs` first stars the rotation, then does some other work (tries to publish new key on ledger), and only after that applies the rotation.

Reply:
Changed.